app/src/org/commcare/logic/AndroidFormController.java

```diff
@@ -20,13 +21,17 @@ public class AndroidFormController extends FormController implements PendingCall
     private boolean wasPendingCalloutCancelled;
     private FormIndex formIndexToReturnTo = null;
     private boolean formCompleteAndSaved = false;
+    private InterruptedFormState restoredFormSession;
 
     private FormAnalyticsHelper formAnalyticsHelper;
 
-    public AndroidFormController(FormEntryController fec, boolean readOnly, FormIndex formIndex) {
+    public AndroidFormController(FormEntryController fec, boolean readOnly, InterruptedFormState savedFormSession) {
         super(fec, readOnly);
         formAnalyticsHelper = new FormAnalyticsHelper();
-        formIndexToReturnTo = formIndex;
+        this.restoredFormSession = savedFormSession;
+        if (savedFormSession !=null ){
+            formIndexToReturnTo = savedFormSession.getFormIndex();
+        }
```

Comment:
think we can remove global reference to `formIndexToReturnTo` and just derive it from `restoredFormSession` when needed

Reply:
`formIndexToReturnTo` [is nullified after its first use](https://github.com/dimagi/commcare-android/blob/2b994dbd9b7a91ca473199b04397e6f27cc0fbcf/app/src/org/commcare/logic/AndroidFormController.java#L76C14-L76C33), which makes sense because we are supposed to return to that FormIndex only when loading the form after a session expiration or app kill. Am I missing something here?

Reply:
makes sense. no need for a change here.